lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/ICloudBlockBlob.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureBlob.Block
+{
+    /// <summary>
+    /// A proxy interface for <see cref="CloudBlockBlob"/>,
+    /// mainly used for unit testing purposes.
+    /// </summary>
+    internal interface ICloudBlockBlob
+    {
+        ICloudBlob Blob { get; }
+
+        BlobProperties Properties { get; }
+
+        bool Exists();
+
+        void Delete();
+
+        void DeleteIfExists();
+
+        Task<string> StartCopyFromBlobAsync(Uri source);
```

Comment:
Sadly, all methods available in the 4.3.0 NuGet are obsolete in the 6.1.0 NuGet, and there are no non-obsolete methods. The ``async`` version is the only available version here.